confluent_kafka/avro/avro_producer.py

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env python
+#
+# Copyright 2016 Confluent Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import logging
+
+from confluent_kafka.avro import ClientError
+from confluent_kafka.avro.cached_schema_registry_client import CachedSchemaRegistryClient
+from confluent_kafka.avro.serializer import SerializerError
+from confluent_kafka.avro.serializer.message_serializer import MessageSerializer
+
+log = logging.getLogger(__name__)
+
+
+class AvroProducer(object):
+    '''
+        Kafka Producer client which does avro schema encoding to messages.
+        Handles schema registration, Message serialization.
+
+        Constructor takes below parameters
+
+        @:param: producer: confluent_kafka.Producer object
+        @:param: schema_registry_url: Schema registry URL
+        @:param: default_key_schema: Optional avro schema for key
+        @:param: default_value_schema: Optional avro schema for value
+    '''
+
+    def __init__(self, producer, schema_registry_url, default_key_schema=None,
+                 default_value_schema=None):  # real signature unknown; restored from __doc__
+        self._producer = producer
+        self._serializer = MessageSerializer(CachedSchemaRegistryClient(url=schema_registry_url))
+        self.key_schema = default_key_schema
+        self.value_schema = default_value_schema
+
+    def produce(self, **kwargs):
+        '''
+            Sends message to kafka by encoding with specified avro schema
+            @:param: topic: topic name
+            @:param: value: A dictionary object
+            @:param: value_schema : Avro schema for value
+            @:param: key: A dictionary object
+            @:param: key_schema : Avro schema for key
+            @:exception: SerializerError
+        '''
+        # get schemas from  kwargs if defined
+        key_schema = kwargs.pop('key_schema', None)
```

Comment:
You can avoid the extra logic below for defaults by just doing `key_schema = kwargs.pop('key_schema', self.key_schema)`